// office-ui-fabric-react/lib/components/Button/CompoundButton/CompoundButton.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var BaseButton_1 = require("../BaseButton");
var styles = require("./CompoundButton.scss");
var CompoundButton = (function (_super) {
    __extends(CompoundButton, _super);
    function CompoundButton() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.classNames = {
            base: 'ms-Button',
            variant: 'ms-Button--compound',
            description: styles.description,
            flexContainer: styles.flexContainer,
            icon: null,
            isDisabled: styles.isDisabled,
            isEnabled: styles.isEnabled,
            label: styles.label,
            root: styles.root
        };
        return _this;
    }
    return CompoundButton;
}(BaseButton_1.BaseButton));
exports.CompoundButton = CompoundButton;
