// office-ui-fabric-react/lib-amd/components/Button/BaseButton.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
define(["require", "exports", "react", "../../Utilities", "../../common/DirectionalHint", "../../ContextualMenu", "./BaseButton.scss"], function (require, exports, React, Utilities_1, DirectionalHint_1, ContextualMenu_1, styles) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var BaseButton = (function (_super) {
        __extends(BaseButton, _super);
        function BaseButton(props, rootClassName, deprecationMap) {
            var _this = _super.call(this, props) || this;
            _this.classNames = {
                base: 'ms-Button',
                variant: '',
                isEnabled: '',
                isDisabled: ''
            };
            _this._warnDeprecations({ 'rootProps': null });
            _this._labelId = Utilities_1.getId();
            _this._descriptionId = Utilities_1.getId();
            _this._ariaDescriptionId = Utilities_1.getId();
            _this.state = {
                menuProps: null
            };
            return _this;
        }
        BaseButton.prototype.render = function () {
            var _a = this.props, description = _a.description, ariaLabel = _a.ariaLabel, ariaDescription = _a.ariaDescription, href = _a.href, disabled = _a.disabled;
            var _b = this, _ariaDescriptionId = _b._ariaDescriptionId, _labelId = _b._labelId, _descriptionId = _b._descriptionId;
            var renderAsAnchor = !!href;
            var tag = renderAsAnchor ? 'a' : 'button';
            var nativeProps = Utilities_1.getNativeProps(Utilities_1.assign({}, this.props.rootProps, this.props), renderAsAnchor ? Utilities_1.anchorProperties : Utilities_1.buttonProperties, [
                'disabled' // Let disabled buttons be focused and styled as disabled.
            ]);
            // Check for ariaDescription, description or aria-describedby in the native props to determine source of aria-describedby
            // otherwise default to null.
            var ariaDescribedBy;
            if (ariaDescription) {
                ariaDescribedBy = _ariaDescriptionId;
            }
            else if (description) {
                ariaDescribedBy = _descriptionId;
            }
            else if (nativeProps['aria-describedby']) {
                ariaDescribedBy = nativeProps['aria-describedby'];
            }
            else {
                ariaDescribedBy = null;
            }
            var buttonProps = Utilities_1.assign(nativeProps, {
                className: Utilities_1.css(styles.root, this.props.className, this.classNames.base, this.classNames.variant, this.classNames.root, (_c = {
                        'disabled': disabled
                    },
                    _c[this.classNames.isDisabled] = disabled,
                    _c[this.classNames.isEnabled] = !disabled,
                    _c)),
                ref: this._resolveRef('_buttonElement'),
                'disabled': disabled,
                'aria-label': ariaLabel,
                'aria-labelledby': ariaLabel ? null : _labelId,
                'aria-describedby': ariaDescribedBy,
                'aria-disabled': disabled
            });
            // Override onClick if contextualMenuItems passed in. Eventually allow _onToggleMenu to
            // be assigned to split button click if onClick already has a value
            if (this.props.menuProps) {
                Utilities_1.assign(buttonProps, { 'onClick': this._onToggleMenu });
            }
            return this.onRenderContent(tag, buttonProps);
            var _c;
        };
        BaseButton.prototype.focus = function () {
            if (this._buttonElement) {
                this._buttonElement.focus();
            }
        };
        BaseButton.prototype.onRenderContent = function (tag, buttonProps) {
            var _a = this.props, _b = _a.onRenderMenu, onRenderMenu = _b === void 0 ? this._onRenderMenu : _b, _c = _a.onRenderMenuIcon, onRenderMenuIcon = _c === void 0 ? this._onRenderMenuIcon : _c, menuProps = _a.menuProps, menuIconName = _a.menuIconName;
            return React.createElement(tag, buttonProps, React.createElement('div', { className: Utilities_1.css(this.classNames.base + '-flexContainer', styles.flexContainer, this.classNames.flexContainer) }, this.onRenderIcon(), this.onRenderLabel(), this.onRenderDescription(), this.onRenderAriaDescription(), this.onRenderChildren(), (menuProps || menuIconName || this.props.onRenderMenuIcon) && onRenderMenuIcon(this.props, this._onRenderMenuIcon), this.state.menuProps && onRenderMenu(menuProps, this._onRenderMenu)));
        };
        BaseButton.prototype.onRenderIcon = function () {
            var icon = this.props.icon;
            return icon && (React.createElement("span", { className: Utilities_1.css(this.classNames.base + "-icon", this.classNames.icon) },
                React.createElement("i", { className: "ms-Icon ms-Icon--" + icon })));
        };
        BaseButton.prototype.onRenderLabel = function () {
            var _a = this.props, children = _a.children, text = _a.text;
            // For backwards compat, we should continue to take in the text content from children.
            if (text === undefined && typeof (children) === 'string') {
                text = children;
            }
            return text && (React.createElement("span", { className: Utilities_1.css(this.classNames.base + "-label", this.classNames.label), id: this._labelId }, text));
        };
        BaseButton.prototype.onRenderChildren = function () {
            var children = this.props.children;
            // If children is just a string, either it or the text will be rendered via onRenderLabel
            // If children is another component, it will be rendered after text
            if (typeof (children) === 'string') {
                return null;
            }
            return children;
        };
        BaseButton.prototype.onRenderDescription = function () {
            var description = this.props.description;
            // ms-Button-description is only shown when the button type is compound.
            // In other cases it will not be displayed.
            return description ? (React.createElement("span", { className: Utilities_1.css(this.classNames.base + "-description", this.classNames.description), id: this._descriptionId }, description)) : (null);
        };
        BaseButton.prototype.onRenderAriaDescription = function () {
            var ariaDescription = this.props.ariaDescription;
            // If ariaDescription is given, descriptionId will be assigned to ariaDescriptionSpan,
            // otherwise it will be assigned to descriptionSpan.
            return ariaDescription ? (React.createElement("span", { className: styles.screenReaderOnly, id: this._ariaDescriptionId }, ariaDescription)) : (null);
        };
        BaseButton.prototype._onRenderMenuIcon = function (props) {
            var _a = props.menuIconName, menuIconName = _a === void 0 ? 'ChevronDown' : _a;
            return (React.createElement("span", { className: Utilities_1.css(this.classNames.base + "-icon", this.classNames.menuIcon) },
                React.createElement("i", { className: "ms-Icon ms-Icon--" + menuIconName })));
        };
        BaseButton.prototype._onRenderMenu = function (menuProps) {
            return (React.createElement(ContextualMenu_1.ContextualMenu, { className: Utilities_1.css('ms-BaseButton-menuHost'), isBeakVisible: true, directionalHint: DirectionalHint_1.DirectionalHint.bottomLeftEdge, items: menuProps.items, target: this._buttonElement, labelElementId: this._labelId, onDismiss: this._onToggleMenu }));
        };
        BaseButton.prototype._onToggleMenu = function () {
            var menuProps = this.props.menuProps;
            var currentMenuProps = this.state.menuProps;
            this.setState({ menuProps: currentMenuProps ? null : menuProps });
        };
        return BaseButton;
    }(Utilities_1.BaseComponent));
    __decorate([
        Utilities_1.autobind
    ], BaseButton.prototype, "_onRenderMenuIcon", null);
    __decorate([
        Utilities_1.autobind
    ], BaseButton.prototype, "_onRenderMenu", null);
    __decorate([
        Utilities_1.autobind
    ], BaseButton.prototype, "_onToggleMenu", null);
    exports.BaseButton = BaseButton;
});
