// react-cartographer/lib/lib/bingMap.js
/**
 * Copyright 2015, Yahoo! Inc.
 * Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
 */

'use strict';

/*
 * Bing MapImage Service
 * @class BingMapService
 */

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _config = require('../../config/config.json');

var _url = require('url');

var _url2 = _interopRequireDefault(_url);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var _ = {
    // Collection
    pick: require('lodash/pickBy'),
    identity: require('lodash/identity')
};

var BingMapService = function () {
    function BingMapService() {
        _classCallCheck(this, BingMapService);
    }

    _createClass(BingMapService, [{
        key: 'getMap',


        /**
         * Get the map location details for the address provided in the params
         *
         * @param {Object} params
         * @returns {{mapId: (*|string|string|string|string), data: {locationLink: *, locationText: string}}}
         */
        value: function getMap(params) {
            var locationText = void 0;
            var location = void 0;
            var pushpin = void 0;
            var url = void 0;

            if (isFinite(params.longitude) && isFinite(params.latitude)) {
                pushpin = location = locationText = [params.latitude, params.longitude].join(',');
                if (isFinite(params.zoom)) {
                    location = [location, params.zoom].join('/');
                }
            } else {
                location = locationText = [params.line1, params.line2, params.line3].join(',');
            }

            url = _url2.default.format({
                protocol: _config.bing.protocol,
                hostname: _config.bing.host,
                pathname: _config.bing.path + encodeURIComponent(location),
                query: _.pick({
                    mapSize: [params.width, params.height].join(','),
                    key: params.providerKey,
                    pushpin: params.withoutMarker ? undefined : pushpin
                }, _.identity)
            });

            return {
                mapId: params.mapId,
                data: {
                    locationLink: url,
                    locationText: locationText
                }
            };
        }
    }, {
        key: 'name',
        get: function get() {
            return _config.bing.name;
        }
    }]);

    return BingMapService;
}();

exports.default = BingMapService;
