// react-cartographer/lib/lib/mapQuestMap.js
/**
 * Copyright 2014, Yahoo! Inc.
 * Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
 */

'use strict';

/*
 * Bing MapImage Service
 * @class BingMapService
 */

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _config = require('../../config/config.json');

var _url = require('url');

var _url2 = _interopRequireDefault(_url);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

var _ = {
    // Collection
    pick: require('lodash/pickBy'),
    identity: require('lodash/identity')
};

var MapQuestService = function () {
    function MapQuestService() {
        _classCallCheck(this, MapQuestService);
    }

    _createClass(MapQuestService, [{
        key: 'getMap',

        /**
         * Get the map location details for the address provided in the params
         *
         * @param {Object} params
         * @returns {{mapId: (*|string|string|string|string), data: {locationLink: *, locationText: string}}}
         */
        value: function getMap(params) {
            var location = [params.line1, params.line2, params.line3].join(',');
            var url = _url2.default.format({
                protocol: _config.mapQuest.protocol,
                hostname: _config.mapQuest.host,
                pathname: _config.mapQuest.path,
                query: _.pick({
                    key: params.providerKey,
                    size: [params.width, params.height].join(','),
                    type: _config.mapQuest.mapType,
                    showicon: _config.mapQuest.icon,
                    location: location
                }, _.identity)
            });

            return {
                mapId: params.mapId,
                data: {
                    locationLink: url,
                    locationText: location
                }
            };
        }
    }]);

    return MapQuestService;
}();

exports.default = MapQuestService;
