// office-ui-fabric-react/lib/components/Button/CommandButton/CommandButton.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var BaseButton_1 = require("../BaseButton");
var styles = require("./CommandButton.scss");
var CommandButton = (function (_super) {
    __extends(CommandButton, _super);
    function CommandButton() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.classNames = {
            base: 'ms-Button',
            variant: 'ms-Button--command',
            icon: styles.icon,
            menuIcon: styles.icon,
            isDisabled: styles.isDisabled,
            isEnabled: styles.isEnabled,
            label: styles.label,
            root: styles.root,
            flexContainer: styles.flexContainer
        };
        return _this;
    }
    CommandButton.prototype.onRenderDescription = function () { return null; };
    return CommandButton;
}(BaseButton_1.BaseButton));
exports.CommandButton = CommandButton;
